src/viur/core/bones/relational.py

```diff
@@ -1105,27 +1105,28 @@ def createRelSkelFromKey(self, key: t.Union[str, "db.Key"], rel: dict | None = N
         relation data.
 
         :param Union[str, db.Key] key: The database key of the entity for which a relSkel instance is to be created.
-        :param Union[dict, None]rel: Optional relation data to be included in the resulting dictionary. Default is None.
 
         :return: A dictionary containing a reference skeleton and optional relation data.
         :rtype: dict
         """
 
-        key = db.keyHelper(key, self.kind)
-        entity = db.Get(key)
-        if not entity:
-            logging.error(f"Key {key} not found")
+        if not all(db_objs := db.Get([db.keyHelper(value[0], self.kind) for value in key_rel_list])):
             return None
-        relSkel = self._refSkelCache()
-        relSkel.unserialize(entity)
-        for k in relSkel.keys():
-            # Unserialize all bones from refKeys, then drop dbEntity - otherwise all properties will be copied
-            _ = relSkel[k]
-        relSkel.dbEntity = None
-        return {
-            "dest": relSkel,
-            "rel": rel or None
-        }
+        res_rel_skels = []
+        for i, db_obj in enumerate(db_objs):
+            dest_skel = self._refSkelCache()
+            dest_skel.unserialize(db_obj)
+            for k in dest_skel.keys():
+                # Unserialize all bones from refKeys, then drop dbEntity - otherwise all properties will be copied
+                _ = dest_skel[k]
+            dest_skel.dbEntity = None
+            res_rel_skels.append(
+                {
+                    "dest": dest_skel,
+                    "rel": key_rel_list[i][1] or None
+                }
+            )
```

Comment:
`i` and `k` are very bad variable names. Furthermore `zip()` is more pythonic in this case.

```suggestion
        for (key, rel), db_obj in zip(key_rel_list, db_objs):
            dest_skel = self._refSkelCache()
            dest_skel.unserialize(db_obj)
            for bone_name in dest_skel:
                # Unserialize all bones from refKeys, then drop dbEntity - otherwise all properties will be copied
                _ = dest_skel[bone_name]
            dest_skel.dbEntity = None
            res_rel_skels.append(
                {
                    "dest": dest_skel,
                    "rel": rel or None
                }
            )
```